src/components/forms/JurisdictionMetadata/index.tsx

```diff
@@ -0,0 +1,230 @@
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import Papaparse from 'papaparse';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import { toast } from 'react-toastify';
+import { Button, Label } from 'reactstrap';
+import { FormGroup } from 'reactstrap';
+import * as Yup from 'yup';
+import {
+  CSV_FILE,
+  FILE,
+  FILE_UPLOADED_SUCCESSFULLY,
+  JURISDICTION_ID,
+  JURISDICTION_METADATA,
+  JURISDICTION_NAME,
+  REQUIRED,
+  UPLOAD,
+  UPLOAD_FILE,
+  UPLOADING,
+} from '../../../configs/lang';
+import { HOME_URL, OPENSRP_V1_SETTINGS_ENDPOINT } from '../../../constants';
+import { growl } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+const SUPPORTED_FORMATS = ['text/csv'];
+
+/** yup validation schema for Jurisdiction Metadata Form input */
+export const JurisdictionSchema = Yup.object().shape({
+  file: Yup.mixed()
+    .required(REQUIRED)
+    .test('fileFormat', 'CSV Files only', value => value && SUPPORTED_FORMATS.includes(value.type)),
+});
+
+export interface JurisdictionMetadataFormFields {
+  file: File;
+}
+
+export interface Setting {
+  description: string;
+  label: string;
+  value: string | unknown;
+  key: string;
+}
+
+export interface SettingConfiguration {
+  type: string;
+  identifier: string;
+  providerId: string;
+  locationId: string;
+  settings: Setting[];
+  teamId: string;
+}
+
+export interface JurisdictionMetadataFormProps {
+  disabledFields: string[];
+  OpenSRPService: new (...args: any[]) => any;
```

Comment:
Why do we need `any` here?  I think we should use an actual type.  We already know the parameters that this class takes, and if all else fails we don't have to make the prop be an instantiated object, it could just be the class that is then instantiated later.

Reply:
We are not passing the parameters here, It's just a placeholder for the parameters in the super. Also I'm doing it here for ease of testing

Reply:
I still don't agree with the weakened typing.  I mean, OpenSRP service is already fairly well typed - we are removing all the goodness that comes with that in this move.  Is there a problem with it being typed as `OpenSRPService: typeof OpenSRPService`?

Reply:
Can we please change the name of this prop to something like `serviceClass`.  Basically I prefer that we avoid re-using `OpenSRPService` - which is the name of the import at the top.  It makes for less easier to read code, and higher chances of using the wrong variable.